src/assessments/assessment-builder.tsx

```diff
@@ -234,54 +234,56 @@ export class AssessmentBuilder {
+        requirements: Requirement[],
+    ): (requirement: string) => VisualizationInstanceProcessorCallback<PropertyBags, PropertyBags> {
+        return (requirementKey: string): VisualizationInstanceProcessorCallback<PropertyBags, PropertyBags> => {
+            const requirementConfig = AssessmentBuilder.getRequirementConfig(requirements, requirementKey);
+            if (requirementConfig == null || requirementConfig.visualizationInstanceProcessor == null) {
                 return VisualizationInstanceProcessor.nullProcessor;
             }
-            return stepConfig.visualizationInstanceProcessor;
+            return requirementConfig.visualizationInstanceProcessor;
         };
     }
 
-    private static getSwitchToTargetTabOnScan(steps: Requirement[]): (testStep: string) => boolean {
-        return (testStep: string): boolean => {
-            const stepConfig = AssessmentBuilder.getStepConfig(steps, testStep);
-            if (stepConfig == null || stepConfig.switchToTargetTabOnScan == null) {
+    private static getSwitchToTargetTabOnScan(requirements: Requirement[]): (requirementKey: string) => boolean {
+        return (requirement: string): boolean => {
+            const requirementConfig = AssessmentBuilder.getRequirementConfig(requirements, requirement);
+            if (requirementConfig == null || requirementConfig.switchToTargetTabOnScan == null) {
                 return false;
             }
-            return stepConfig.switchToTargetTabOnScan;
+            return requirementConfig.switchToTargetTabOnScan;
         };
     }
 
-    private static getInstanceIdentifier(steps: Requirement[]): (testStep: string) => (instance: UniquelyIdentifiableInstances) => string {
-        return (testStep: string) => {
-            const stepConfig = AssessmentBuilder.getStepConfig(steps, testStep);
-            if (stepConfig == null || stepConfig.generateInstanceIdentifier == null) {
+    private static getInstanceIdentifier(
+        requirements: Requirement[],
+    ): (requirement: string) => (instance: UniquelyIdentifiableInstances) => string {
+        return (requirementKey: string) => {
+            const requirementConfig = AssessmentBuilder.getRequirementConfig(requirements, requirementKey);
+            if (requirementConfig == null || requirementConfig.generateInstanceIdentifier == null) {
                 return InstanceIdentifierGenerator.defaultHtmlSelectorIdentifier;
             }
-            return stepConfig.generateInstanceIdentifier;
+            return requirementConfig.generateInstanceIdentifier;
         };
     }
 
-    private static BuildStepsReportDescription(steps: Requirement[]): void {
-        steps.forEach(step => {
-            step.renderReportDescription = () => {
-                const descriptionCopy = _.cloneDeep(step.description);
+    private static BuildRequirementReportDescription(requirements: Requirement[]): void {
```

Comment:
this name should be camel-case, not pascal-case right? i.e **b**uildRequirementReportDescription

Reply:
updated!